src/Microsoft.AspNetCore.Mvc.Core/ApiExplorer/ApiConventionMatcher.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ApiExplorer
+{
+    internal static class ApiConventionMatcher
```

Comment:
You were right - https://github.com/aspnet/Mvc/pull/7939#discussion_r197300235. Not exactly a provider, but it makes it easy to have a `ApiConventionMatcher` that's mirrored by `SymbolApiConventionMatcher`